frontend/packages/dev-console/src/components/import/git/SourceSecretSelector.tsx

```diff
@@ -45,6 +46,9 @@ const SourceSecretSelector: React.FC<{
         save: handleSave,
         secretType: SecretTypeAbstraction.source,
       });
+    } else if (key === CLEAR_SOURCE_SECRET) {
+      setFieldValue(`${fieldPrefix}git.secret`, null);
+      setFieldTouched(`${fieldPrefix}git.dir`);
```

Comment:
Is this needed to trigger the api call or why do we set the contextDir to touched here?

Reply:
Previously, when we were removing the secret by selecting the "No Secret" option from the Dropdown, the secret was getting removed, but there was no validation check happening in the GIT URL field, which was normally happening when we were either adding or changing the secret. 
Therefore the replicate the behavior in case of `No secret` too, I have added the `setFieldTouched`